commit 39aad99285cd9bd845f2f91601d8e8c3216c0d0f (refs/changes/00/4000/4)
Author: Pedro Palazon Candel <pedro@joyent.com>
Date:   2018-05-31T20:45:56+02:00 (1 year, 4 months ago)
    
    TRITON-395 Drop sdcadm support for manatee v1.0
    Reviewed by: Marsell Kukuljevic <marsell@joyent.com>
    Approved by: Marsell Kukuljevic <marsell@joyent.com>

diff --git a/etc/defaults.json b/etc/defaults.json
index b49f01e..ae969c2 100644
--- a/etc/defaults.json
+++ b/etc/defaults.json
@@ -65,7 +65,7 @@
         "vmapi": "20140710T171616Z",
         "moray": "20140710T181209Z",
         "adminui": "20140718T210958Z",
-        "manatee": "20140721T221128Z",
+        "manatee": "20141225T181254Z",
         "sdc": "20140725T191132Z",
         "cloudapi": "20140731T175858Z",
         "ufds": "20140728T173339Z"
diff --git a/lib/procedures/update-manatee-v2.js b/lib/procedures/update-manatee-v2.js
index 2791dca..4da6faa 100644
--- a/lib/procedures/update-manatee-v2.js
+++ b/lib/procedures/update-manatee-v2.js
@@ -206,7 +206,6 @@ UpdateManateeV2.prototype.execute = function manateev2Execute(opts, cb) {
         var sapiUUID;
         var morayVms;
         var wfVms;
-        var version;
         var pastSAPI224 = false;
 
         // TOOLS-975: As a workaround for any manatee instance affected by
@@ -351,7 +350,7 @@ UpdateManateeV2.prototype.execute = function manateev2Execute(opts, cb) {
         }
 
         function freeze(_, next) {
-            if (!arg.HA || version === '1.0.0' || isFrozen) {
+            if (!arg.HA || isFrozen) {
                 next();
                 return;
             }
@@ -838,28 +837,6 @@ UpdateManateeV2.prototype.execute = function manateev2Execute(opts, cb) {
                 }, next);
             },
 
-            function getManateeAdmVersion(_, next) {
-                progress('Checking manatee-adm version');
-                common.manateeAdmRemote({
-                    server: primary.server_uuid,
-                    vm: primary.zoneId,
-                    cmd: 'version',
-                    log: log
-                }, function (err, stdou, stder) {
-                    if (err) {
-                        next(err);
-                        return;
-                    }
-                    // Not implemented until 2.0.0
-                    if (stder) {
-                        version = '1.0.0';
-                    } else {
-                        version = stdou;
-                    }
-                    next();
-                });
-            },
-
             // --------------- HA only --------------------------------------
             function verifyFullHA(_, next) {
                 if (!arg.HA) {
@@ -884,7 +861,6 @@ UpdateManateeV2.prototype.execute = function manateev2Execute(opts, cb) {
             },
 
             function reprovisionAsyncPeers(_, next) {
-                var backfillDone = false;
                 function reprovisionAsyncPeer(peer, nextPeer) {
                     vasync.pipeline({
                         arg: peer,
@@ -941,26 +917,6 @@ UpdateManateeV2.prototype.execute = function manateev2Execute(opts, cb) {
                                 }, nextStep);
                             },
 
-                            function asyncStateBackfill(p, nextStep) {
-                                if (arg.updated_vms.indexOf(p.zoneId) !== -1 ||
-                                    !arg.HA || version !== '1.0.0' ||
-                                    backfillDone) {
-                                    nextStep();
-                                    return;
-                                }
-
-                                progress('Backfilling cluster state');
-                                common.manateeAdmRemote({
-                                    server: p.server_uuid,
-                                    vm: p.zoneId,
-                                    cmd: 'state-backfill -y',
-                                    log: log
-                                }, function (err) {
-                                    backfillDone = true;
-                                    nextStep(err);
-                                });
-                            },
-
                             function waitForHA(p, nextStep) {
                                 if (arg.updated_vms.indexOf(p.zoneId) !== -1 ||
                                     !arg.HA) {
@@ -981,6 +937,7 @@ UpdateManateeV2.prototype.execute = function manateev2Execute(opts, cb) {
                     inputs: asyncs || [],
                     func: reprovisionAsyncPeer
                 }, next);
+
             },
 
             function freezeBeforeSync(_, next) {
@@ -1105,19 +1062,6 @@ UpdateManateeV2.prototype.execute = function manateev2Execute(opts, cb) {
             },
 
             // ----------- Again, no-HA only ----------------------------------
-            function noHAStateBackfill(_, next) {
-                if (arg.HA || version !== '1.0.0') {
-                    next();
-                    return;
-                }
-                progress('Backfilling cluster state');
-                common.manateeAdmRemote({
-                    server: primary.server_uuid,
-                    vm: primary.zoneId,
-                    cmd: 'state-backfill -y',
-                    log: log
-                }, next);
-            },
 
             function waitForPrimaryPG(_, next) {
                 if (arg.HA) {
